### DISCUSSION TITLE
Imagesharp throwing System.MissingMethodException

### DISCUSSION THREAD

**Discussion by Maddy-176** [SixLabors/ImageSharp#2746] [Q&A]
I am trying to use pdfsharpcore and SixLabours.imageshrap however I am consistently getting the issue _System.MissingMethodException: Method not found: 'SixLabors.ImageSharp.Image`1<!!0> SixLabors.ImageSharp.Image.Load(System.IO.Stream, SixLabors.ImageSharp.Formats.IImageFormat ByRef)_

The error gets throws consistently in the last line of this code
```
Stream imageStream= GetImage(imageUrl);

Image image= Image.Load(imageStream);
image.Mutate(img=>img.Resize(width,height));
MemoryStream ms = new MemoryStream();
image.SaveAsPng(ms);
return XImage.FromStream(()=>new MemoryStream(ms.ToArray());
```
Based on the recommendtion [here](https://stackoverflow.com/questions/78549214/imagesharp-throwing-system-missingmethodexception) i downgraded my pdfsharpcore version and imagesharp version . Below are the versions used:
pdfsharpcore: 1.0.3
imagesharp: 1.0.0-beta006  
.net version: 6.0

Can anyone suggest how can we fix this? The issue s only happening when I am trying to render images.

**Comment by antonfirsov** [comment]
PdfSharpCore depends on [ImageSharp 1.0.4](https://www.nuget.org/packages/PdfSharpCore/), so the beta version is too low. Note that ImageSharp 1.* is out of support and full of security holes which renders PdfSharpCore an insecure lib too.

If you want to continue using that lib regardless, then - instead of adjusting the versions in the same project/process - my recommendation would be to move your image processing logic into a separate process and use the latest ImageSharp there .

**Reply by Maddy-176** [reply]
Actually I cannot move the image processing logic seperately right now. Also if I am using the latest stable version of both image sharp and pdfsharpcore I am still getting this issue

**Reply by tocsoft** [reply]
@Maddy-176 as I pointed out on Stack Overflow last week the latest version of `PdfSharpCore` is just not compatible with the latest version of `ImageSharp` and there is nothing we can do to make it work for you. As outlined over on Stack Overflow you need to talk to the maintainers of `PdfSharpCore` (which is not us and is on affiliated with us in anyway) to get them to update to an up-to-date version of `ImageSharp`, or use an out of date and unsupported version of ImageSharp (v1.0.4 specifically which both Anton and myself have pointed out) with all the inherent risks involved.

If you don't downgrade `ImageSharp` (which I wouldn't really recommend doing anyway) and you can't get the `PdfSharpCore` team to bother updating, then your best bet would to grab the the `PdfSharpCore` source code and compile it yourself as part of your project (you will then be able to update the version of `ImageSharp` used by it yourself then). 

Please note however that taking that approach isn't something we can really help or support you doing as non of our team have any involvement or real experience in the internals of `PdfSharpCore`.

**Reply by TashievEmir** [reply]
@tocsoft is your answer can be also related with SourceAFIS library? I mean when I using SourceAFIS library i guess it try to use some your code and i get the same error?

**Reply by JimBobSquarePants** [reply]
Yes. See their dependency list on NuGet. They’re binary compatible with version 2.1.3. 

https://www.nuget.org/packages/SourceAFIS/#dependencies-body-tab

**Comment by tiesont** [comment]
Assuming the core issue here is getting an image snapshot for a page from a PDF and then resizing that image, I've successfully used https://github.com/GowenGit/docnet to do that in a project that started as .NET6 and is currently .NET8. 

You can get the image as a BRGA byte array (there's a `GetImage()` method defined for it's `IPageReader` interface). ImageSharp 3.x has had no issue loading and manipulating that byte array.

I do concur that this is not really on-topic for ImageSharp, but I figured I'd mention the above if it's at all helpful.
